rdkcentral/Dobby
# Allow building Dobby with and without systemd

### Description
Support Dobby on S6 platforms.

When USE_SYSTEMD is ON, use `sd-bus` libraries and provide systemd integration
When USE_SYSTEMD is OFF, use `libdbus` libraries and remove systemd integration

Using systemd is still recommended if it is available as the `sd-bus` implementation of IPCService is more reliable and easier to debug in case of issues.

### Test Procedure
* Build Dobby with `-DUSE_SYSTEMD=OFF` on a platform that does not include systemd. Dobby should build without errors
* Dobby should run and launch containers without errors on such platforms.

Note that if Dobby is launched without systemd, it *may* be necessary to add a custom dbus config xml file in `/etc/dbus-1/system.d/org.rdk.dobby.conf`

### Type of Change

### Requires Bitbake Recipe changes?
